app/login/page.tsx

```diff
@@ -17,52 +42,177 @@ const Login = () => {
       [field]: event.target.value,
     });
   };
-
+   // Function to toggle password visibility
+  const handleTogglePasswordVisibility = () => {
+    setFormData({
+      ...formData,
+      showPassword: !formData.showPassword,
+    });
+  };
+  // Function to handle form submission
   const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
     // Handle login logic here, e.g., send data to the server
     console.log(formData);
   };
 
   return (
-    <Container maxWidth="sm">
```

Comment:
Don't replace code snippet like this. You have to seamlessly integrate your work to the existing code without breaking the existing functionality.